python-sdk/src/astro/custom_backend/astro_custom_backend.py

```diff
@@ -31,11 +32,12 @@ def serialize_value(value: Any, **kwargs) -> Any:  # type: ignore[override]
         if DATAFRAME_STORAGE_CONN_ID or STORE_DATA_LOCAL_DEV:
             value = serialize(value)
         else:
-            warnings.warn(
-                "Warning: Since you have not provided a remote object storage conn_id, we are defaulting to"
-                "storing your data in the Metadata DB. This does not scale well and can cause degradation to"
-                "your airflow DB. Please set the AIRFLOW__ASTRO__XCOM_STORAGE_CONN_ID and "
-                "AIRFLOW__ASTRO__XCOM_STORAGE_URL variables"
+            raise AirflowException(
+                "Since you have not provided a remote object storage conn_id, we are defaulting to "
+                "storing your data in the Metadata DB. This does not scale well and can cause degradation to "
+                "your airflow DB. Please set the AIRFLOW__ASTRO_SDK__XCOM_STORAGE_CONN_ID and "
+                "AIRFLOW__ASTRO_SDK__XCOM_STORAGE_URL variables "
+                "(or you can set AIRFLOW__ASTRO_SDK__STORE_DATA_LOCAL_DEV for local development)"
```

Comment:
Where do we catch this exception? Because it is saying that we default to sth...

Reply:
fixed